sdk/python/kfp/local/importer_handler.py

```diff
@@ -90,13 +133,53 @@ def run_importer(
     )
     ArtifactCls = get_artifact_class_from_schema_title(
         executor_spec.importer.type_schema.schema_title)
-    outputs = {
-        'artifact': ArtifactCls(
-            name='artifact',
-            uri=uri,
-            metadata=metadata,
-        )
-    }
+    artifact = ArtifactCls(
+        name='artifact',
+        uri=uri,
+        metadata=metadata,
+    )
+
+    if executor_spec.importer.download_to_workspace:
+        parsed_uri = urlparse(uri)
+        scheme = parsed_uri.scheme
+        if scheme not in ('', 'file'):
+            warnings.warn(
+                "download_to_workspace only supports local file paths when running locally. Skipping workspace copy for '%s'."
+                % uri)
+        else:
+            source_path = parsed_uri.path if scheme == 'file' else uri
+            try:
+                workspace_root = _get_workspace_root()
+                cfg = config.LocalExecutionConfig.instance
+                used_base = 'workspace'
+                try:
+                    host_path = _copy_local_artifact_to_workspace(
+                        source_path, workspace_root, component_name)
+                except PermissionError:
+                    if isinstance(cfg.runner, config.DockerRunner):
+                        # fallback to pipeline_root for Docker
+                        used_base = 'pipeline_root'
+                        host_path = _copy_local_artifact_to_workspace(
+                            source_path, cfg.pipeline_root, component_name)
```

Comment:
I thought workspace_root was mounted as a volume for the DockerRunner. Is this fallback needed?